File: webapi_v2/java/WebaiuiDemo.java
package com.iflytek.aiui.demo;

import java.io.BufferedReader;
import java.io.ByteArrayOutputStream;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.text.ParseException;
import java.util.HashMap;
import java.util.Map;
import org.apache.commons.codec.binary.Base64;
import org.apache.commons.codec.digest.DigestUtils;

/**
 * AIUI WebAPI V2接口调用示例
 * 
 * 运行方法：直接运行 main()
 * 
 * 结果： 控制台输出接口返回值信息
 * 
 * @author iflytek_aiui
 * 
 */
public class WebaiuiDemo {
	private static final String URL = "http://api.xfyun.cn/v2/aiui";
	private static final String APPID = "";
	private static final String API_KEY = "";
	private static final String DATA_TYPE = "audio";
	private static final String SCENE = "main";
	private static final String SAMPLE_RATE = "16000";
	private static final String UNIQUE_ID = "2894c985bf8b1111c6728db79d3479ae5";
	private static final String AUE = "raw";
	private static final String FILE_PATH = "";
	
	public static void main(String[] args) throws IOException,ParseException, InterruptedException{
		Map<String, String> header = buildHeader();
		byte[] dataByteArray = readFile(FILE_PATH);
		String result = httpPost(URL, header, dataByteArray);
		System.out.println(result);		
	}

	private static Map<String, String> buildHeader() throws UnsupportedEncodingException, ParseException {
		String curTime = System.currentTimeMillis() / 1000L + "";
		String param = "{\"aue\":\""+AUE+"\",\"sample_rate\":\""+SAMPLE_RATE+"\",\"unique_id\":\""+UNIQUE_ID+"\",\"data_type\":\""+DATA_TYPE+"\",\"scene\":\""+SCENE+"\"}";		
		String paramBase64 = new String(Base64.encodeBase64(param.getBytes("UTF-8")));
		String checkSum = DigestUtils.md5Hex(API_KEY + curTime + paramBase64);

		Map<String, String> header = new HashMap<String, String>();
		header.put("X-Param", paramBase64);
		header.put("X-CurTime", curTime);
		header.put("X-CheckSum", checkSum);
		header.put("X-Appid", APPID);
		return header;
	}
	
	private static byte[] readFile(String filePath) throws IOException {
		InputStream in = new FileInputStream(filePath);
		ByteArrayOutputStream out = new ByteArrayOutputStream();
		byte[] buffer = new byte[1024 * 4];
		int n = 0;
		while ((n = in.read(buffer)) != -1) {
			out.write(buffer, 0, n);
		}
		byte[] data = out.toByteArray();
		in.close();
		return data;
	}
	
	private static String httpPost(String url, Map<String, String> header, byte[] body) {
		String result = "";
		BufferedReader in = null;
		OutputStream out = null;
		try {
			URL realUrl = new URL(url);
			HttpURLConnection connection = (HttpURLConnection)realUrl.openConnection();
			for (String key : header.keySet()) {
				connection.setRequestProperty(key, header.get(key));
			}
			connection.setDoOutput(true);
			connection.setDoInput(true);
			
			//connection.setConnectTimeout(20000);
			//connection.setReadTimeout(20000);
			try {
				out = connection.getOutputStream();
				out.write(body);
				out.flush();
			} catch (Exception e) {
				e.printStackTrace();
			}
			
			try {
				in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
				String line;
				while ((line = in.readLine()) != null) {
					result += line;
				}
			} catch (Exception e) {
				e.printStackTrace();
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		return result;
	}
}

File: webapi_v2/php/WebaiuiDemo.php
<?php
class Demo{
	function testWebaiui(){
		$URL = "http://openapi.xfyun.cn/v2/aiui";
		$APPID = "";
		$API_KEY = "";		
		$AUTH_ID = "2894c985bf8b1111c6728db79d3479ae";
		$AUE = "raw";
		$SAMPLE_RATE = "16000";
		$DATA_TYPE = "audio";
		$SCENE = "main";
		$RESULT_LEVEL = "complete";
		$LAT = "39.938838";
		$LNG = "116.368624";		
		$FILE_PATH = "";
		
		$Param= array(
			"result_level"=>$RESULT_LEVEL,
			"aue"=>$AUE,
			"scene"=>$SCENE,
			"auth_id"=>$AUTH_ID,
			"data_type"=>$DATA_TYPE,
			"sample_rate"=>$SAMPLE_RATE,
			"lat"=>$LAT,
			"lng"=>$LNG,
		);

		$curTime = time();
		$paramBase64 = base64_encode(json_encode($Param));
		$checkSum = md5($API_KEY.$curTime.$paramBase64);
		
		$headers = array();
		$headers[] = 'X-CurTime:'.$curTime;
		$headers[] = 'X-Param:'.$paramBase64;		
		$headers[] = 'X-CheckSum:'.$checkSum;
		$headers[] = 'X-Appid:'.$APPID;
		
		$fp = fopen($FILE_PATH, "rb");
		$dataArray = fread($fp,filesize($FILE_PATH));
		
		echo $this->https_request($URL, $dataArray, $headers);
		echo "\n";
	}

	function https_request($url, $post_data, $headers) {	
	  $options = array(
		'http' => array(
		  'method' => 'POST',
		  'header' => $headers,
		  'content' => $post_data,
		  'timeout' => 10
		)
	  );
	  $context = stream_context_create($options);
	  $result = file_get_contents($url, false, $context);
		
	  echo $result; 
	}
}
$demo = new Demo();
$demo->testWebaiui();
?>